### PR TITLE
SWC-6923 - Use React implementation for verification rejection dialog

### PR DESCRIPTION
**Pull request by nickgros**

### CODE REVIEW

File: .husky/pre-commit [hunk 1]
```diff
@@ -1,4 +1 @@
-#!/bin/sh
-. "$(dirname "$0")/_/husky.sh"
-
```
Comment (nickgros) [inline .husky/pre-commit:3]:
```
husky - DEPRECATED

Please remove the following two lines from .husky/pre-commit:

#!/usr/bin/env sh
. "$(dirname -- "$0")/_/husky.sh"

They WILL FAIL in v10.0.0
```

File: src/main/resources/portal.properties [hunk 2]
```diff
@@ -2,8 +2,10 @@ org.sagebionetworks.portal.challenge_synapse_id = syn1968135
 org.sagebionetworks.portal.challenge_writeup_synapse_id = syn2027095
 org.sagebionetworks.portal.default_thread_id = 234
 
-org.sagebionetworks.portal.ACT_profile_validation_rejection_reasons_json = syn24977268
+org.sagebionetworks.portal.ACT_profile_validation_rejection_reasons_json = syn24977268
```
Comment (nickgros) [inline src/main/resources/portal.properties]:
```suggestion
org.sagebionetworks.portal.ACT_profile_validation_rejection_reasons_json = syn24977268
```

File: src/main/java/org/sagebionetworks/web/client/widget/verification/VerificationSubmissionWidget.java [hunk 3]
```diff
@@ -425,22 +434,14 @@ public void suspendVerification() {
   }
 
   private void rejectSuspendVerification() {
-    RejectReasonWidget promptModal = ginInjector.getRejectReasonWidget();
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/verification/VerificationSubmissionWidget.java:428]: Can RejectReasonWidget be deleted?

**Review by jay-hodgson — APPROVED** [review]

File: src/main/java/org/sagebionetworks/web/client/widget/verification/VerificationSubmissionWidget.java (hunk 3 above)
Comment (nickgros) [reply src/main/java/org/sagebionetworks/web/client/widget/verification/VerificationSubmissionWidget.java:428]: It's still used by some data access request widgets, though those may also be deletable

**Pull request merged**